### PR TITLE
`iotedge check` improvements

### PR DESCRIPTION
**Pull request by arsing**
- Restructure list of checks per the new spec.

- Allow the user to override the expected latest version of iotedged.
  If this is set, the command does not reach out to the aka.ms URL to
  fetch the expected latest version.

- Allow the user to override the name of the diagnostics image.

- Differentiate between checks with warnings and checks that were skipped
  because a previous check failed.

- Hide skipped tests unless `--verbose` is set.

- Add check that device will not run in quick start mode.

- Downgrade error that Moby is not being used in Windows to a warning.

- Fix check that the edge agent can reach the management URI to handle
  the case where the listening URI is an fd.

- Downgrade error that iotedged is not the latest version to a warning.

- Fix host local time check to handle the case where the host is too
  far behind the SNTP server.

- Fix host local time check to report a warning instead of an error if
  the SNTP server is not reachable.

- Remove checks for the versions of edge agent and edge hub, since
  these should be reported by the portal, etc.

- Remove the check that the Azure IoT Hub AMQP and MQTT ports can be reached
  from the host. Instead check these from within a module container.

- Add check that any ports that edge hub wants to bind to on the host are
  available.

- Use colors when writing the success / warning / error results of
  the checks.

- Add note to default config.yaml about how the connect URIs must match
  the file paths in the systemd socket files.

### CODE REVIEW

**Comment by arsing** [comment]
The "host can complete a TLS handshake with Azure IoT Hub" test adds a dependency on `ssleay32.dll`, which we currently don't ship.

File: edgelet/iotedge/src/check.rs [hunk 1]
```diff
@@ -174,133 +195,210 @@ impl Check {
...
+                        connection_to_iot_hub_container(check, 443)
                     }),
-                    ("can connect to IoT Hub MQTT port", |check| {
-                        connection_to_iot_hub(check, 8883)
+                    ("container can connect to IoT Hub MQTT port", |check| {
+                        connection_to_iot_hub_container(check, 8883)
                     }),
+                    ("edge hub can bind to ports on host", edge_hub_ports_on_host),
+                    (
+                        "container runtime network allows name resolution",
```
Comment (myagley) [inline edgelet/iotedge/src/check.rs]: Is this message correct?

File: edgelet/iotedge/src/check.rs [hunk 2]
```diff
@@ -801,14 +772,252 @@ fn container_local_time(check: &mut Check) -> Result<Option<String>, failure::Er
...
+        max_file: Option<String>,
+
+        #[serde(rename = "max-size")]
+        max_size: Option<String>,
+    }
+
+    let daemon_config_path = if cfg!(windows) {
+        r"C:\ProgramData\iotedge-moby\config\daemon.json"
+    } else {
+        "/etc/docker/daemon.json"
```
Comment (myagley) [inline edgelet/iotedge/src/check.rs]: Should we make these constants?

File: edgelet/iotedge/src/check.rs (hunk 1 above)
Comment (arsing) [reply edgelet/iotedge/src/check.rs]: Venkat suggests "modules on the container runtime network can resolve each other by name".

File: edgelet/iotedge/src/check.rs (hunk 2 above)
Comment (arsing) [reply edgelet/iotedge/src/check.rs]: Done.

File: edgelet/iotedge/src/check.rs (hunk 1 above)
Comment (myagley) [reply edgelet/iotedge/src/check.rs]: Oh wait. I thought this was related to the port binding check. You can disregard my comment.

**Pull request merged**